slsReceiverSoftware/src/BinaryFile.cpp

```diff
@@ -72,7 +72,7 @@ int BinaryFile::WriteToFile(char* buffer, int buffersize, uint64_t fnum, uint32_
 	// check if maxframesperfile = 0 for infinite
 	if ((*maxFramesPerFile) && (numFramesInFile >= (*maxFramesPerFile))) {
 		CloseCurrentFile();
-		CreateFile(fnum);
+		CreateFile(++subFileIndex);
```

Comment:
You don't need to pass the subfleindex as it is accessible by CreateFile. I believe you just need to increment it in CreateFile, but then you will have to reset it to -1 (or increment it here and call CreateFile). Please also edit the comment in the header file for CreateFile stating what fnum is as it is misleading.

Reply:
Ok, since the variable is unsigned I went with the increment option, even though overflow would have worked.